### ISSUE TITLE
[4.0] Impossible to uninstall an extension

### ISSUE THREAD

**Issue by joeforjoomla** [joomla/joomla-cms#24155]
### Steps to reproduce the issue
Uninstall an extension on Joomla 4

### Expected result
Extension is uninstalled correctly

### Actual result
Error 'Too few arguments'  passed to the function 'onExtensionBeforeUninstall' of the new plugin 'extension/finder/finder.php'

### Additional comments
Additionally i ignore the purpose of this plugin

**Comment by infograf768** [comment]
Looks like the code is wrong in `/libraries/src/Installer/Installer.php` line 756-761.
We have 
```
		// Fire the onExtensionBeforeUninstall event.
		PluginHelper::importPlugin('extension');
		Factory::getApplication()->triggerEvent(
			'onExtensionBeforeUninstall',
			array('eid' => $identifier)
		);

```
Looks like it should be
```
		// Fire the onExtensionBeforeUninstall event.
		PluginHelper::importPlugin('extension');
		Factory::getApplication()->triggerEvent(
			'onExtensionBeforeUninstall',
			array('installer' => clone $this, 'eid' => $identifier, 'removed' => $result)
		);
```
Can you test?

The plugin is used to `add or remove common words of a language from com_finder`

**Comment by infograf768** [comment]
btw, which extension do you try to uninstall?

**Comment by joeforjoomla** [comment]
Yes @infograf768 if you pass the correct parameters to the onExtensionBeforeUninstall event obviously it works. The issue at the moment occurs with whatever extension/plugin you are trying to uninstall.
However i still have not clear what's the reason of executing such a plugin to remove common words of a language from com_finder, for example when installing/uninstalling a simple system plugin that has nothing to do with com_finder.
To me this sounds a bit not logical and redundant,

**Comment by infograf768** [comment]
Let me make a patch for the error.
Concerning finder, absolutely no idea.

@Hackwar should know better.

**Comment by joeforjoomla** [comment]
Maybe i've understood the purpose looking better at the code. This is something that relates to language extensions only. IMHO it would be better to check immediately at the beginning of onExtensionAfterInstall, onExtensionBeforeUninstall, etc if the extension is of type language, instead of execute a database query. I see a large margin of improvements in this plugin.

**Comment by infograf768** [comment]
Patch here
https://github.com/joomla/joomla-cms/pull/24159

Closing.
Please create a new issue specifically for the Extension- Finder plugin. I have no idea what is a language extension plugin.

**Comment by infograf768** [comment]
I reopen. The issue may be in the plugin itself. It's the only one though using this event.

**Comment by alikon** [comment]
please test #24162

**Comment by ghost** [comment]
closed as having Pull Request #24162

**Comment by SharkyKZ** [comment]
Test PR #24163 please.
